charts/brigade/templates/gateway-generic-ingress.yaml

```diff
@@ -16,16 +18,29 @@ metadata:
       {{ $key }}: {{ $value | quote }}
     {{- end }}
 spec:
+  {{- if and (.Values.ingress.ingressClassName) ($networkingApiSupportsIngressClassName) }}
+  ingressClassName: {{ .Values.ingress.ingressClassName }}
```

Comment:
Should be `.Values. genericGateway.ingress.ingressClassName`

Reply:
see above